scenes/ui/progress_button/progress_button.gd

```diff
@@ -1,8 +1,14 @@
-extends MarginContainer
-class_name ProgressButton
+class_name ProgressButton extends MarginContainer
 
 const UNPAID_ANIMATION_LENGTH: float = 0.3
 
+@export var sfx_button_down: AudioStream
```

Comment:
I'd like to avoid exports - they seem "scattered" across the project to me, less organized.

Suggestion: use a centralized node i.e. `res://global/autoload/resources.gd` to preload audio resources and then access them by unique id's, same like other resources in the game are handled.

For example, enemy image uses the centralized resources node to provide it's image
```
@export var enemy_image_id: String

func get_enemy_image_texture() -> Resource:
	return Resources.enemy_image.get(enemy_image_id, null)
```

perhaps other game_data resources can provide their audio resources in the same way?

then we can also have default id's set in the resource script, e.g. 
```
@export var audio_mouse_down_id: String = "UI_Simple_5_1"
```

Let me know what you think about this approach. 👁️

Reply:
So if we did it using IDs, we would still have export vars with the ID name, right? I'm not opposed to this to keep things centralized as you mentioned. I am curious how this loads the audio though. I'm not sure if it would load all sfx into memory when the provider is instantiated, I can investigate

As for not having exports on UI, how would we configure having SFX on UI items like buttons? Right now it's just the resource/progress buttons that have sound, but what about when it's the main menu UI or swapping tabs/assigning population, or any other UI elements?

Reply:
Hmm, maybe something like localization provider? 

Each node has its id's for sounds, and requests audio stream from the audio provider using a id string.

Reply:
Good questions - I'll have to think about it. We can leave it as it for now until we find a better way.